### PR TITLE
Certbot with nginx

### PR DESCRIPTION
**Pull request by tonybenoy**
@Yenthe666 @sswapnesh

### CODE REVIEW

**Comment by tonybenoy** [comment]
@izotecarlitos You could possibly test it out

**Comment by Yenthe666** [comment]
Cool! I totally have to spin up a little droplet with a subdomain to test this :D Remind me in like four days if you haven't heard back please.

File: odoo_install.sh [hunk 1]
```diff
@@ -377,4 +376,7 @@ echo "Password superadmin (database): $OE_SUPERADMIN"
 echo "Start Odoo service: sudo service $OE_CONFIG start"
 echo "Stop Odoo service: sudo service $OE_CONFIG stop"
 echo "Restart Odoo service: sudo service $OE_CONFIG restart"
+if [ $INSTALL_NGINX = "True" ]; then
+  echo "Nginx configuration at: /etc/nginx/sites-available/odo
```
Comment (Yenthe666) [inline odoo_install.sh]:
```suggestion
  echo "Nginx configuration at: /etc/nginx/sites-available/odoo
```

Comment (tonybenoy) [reply odoo_install.sh]: @Yenthe666  you reviewed the outdated file

**Comment by tonybenoy** [comment]
@Yenthe666 bump

File: README.md [hunk 2]
```diff
@@ -21,6 +21,11 @@ There are a few things you can configure, this is the most used list:<br/>
 ```OE_SUPERADMIN``` is the master password for this Odoo installation.<br/>
 ```INSTALL_NGINX``` is set to ```False``` by default. Set this to ```True``` if you want to install Nginx.<br/>
 ```WEBSITE_NAME``` Set the website name here for nginx configuration<br/>
+```WEBSITE_NAME``` Set the website name here for nginx configuration<br/>
+```ENABLE_SSL``` Set installs [certbot](https://github.com/certbot/certbot) configures https with Let's Encrypted certificate<br/>
+```ADMIN_EMAIL``` Email is needed to register for Let's Enctryped Registeration. Replace the default place holder with an email of your organisation.<br/>
+```INSTALL_NGINX``` and ```ENABLE_SSL``` must be set to ```True``` and the place holder in ```ADMIN_EMAIL``` must be replaced with a valid email address for certbot installation<br/>
```
Comment (Yenthe666) [inline README.md]: Neatpick: "placeholder" is one word.

File: README.md [hunk 3]
```diff
@@ -21,6 +21,11 @@ There are a few things you can configure, this is the most used list:<br/>
 ```OE_SUPERADMIN``` is the master password for this Odoo installation.<br/>
 ```INSTALL_NGINX``` is set to ```False``` by default. Set this to ```True``` if you want to install Nginx.<br/>
 ```WEBSITE_NAME``` Set the website name here for nginx configuration<br/>
+```WEBSITE_NAME``` Set the website name here for nginx configuration<br/>
+```ENABLE_SSL``` Set installs [certbot](https://github.com/certbot/certbot) configures https with Let's Encrypted certificate<br/>
```
Comment (Yenthe666) [inline README.md]: Can you rephrase this sentence? Its not very clear I believe

File: odoo_install.sh [hunk 4]
```diff
@@ -38,7 +38,10 @@ OE_CONFIG="${OE_USER}-server"
 WEBSITE_NAME="_"
 # Set the default Odoo longpolling port (you still have to use -c /etc/odoo-server.conf for example to use this.)
 LONGPOLLING_PORT="8072"
-
+# Set to "True" to install certbot and have ssl enabled, "False" to use http
+ENABLE_SSL="True"
```
Comment (Yenthe666) [inline odoo_install.sh:42]: Let's set this to `False` by default and maybe add a link to the certbot docs

File: odoo_install.sh [hunk 5]
```diff
@@ -345,6 +348,21 @@ EOF
...
+#--------------------------------------------------
+
+if [ $INSTALL_NGINX = "True" ] && [ $ENABLE_SSL = "True" ] && [ $ADMIN_EMAIL != 'odoo@example.com' ];then
+  sudo add-apt-repository ppa:certbot/certbot -y && sudo apt-get update -y
+  sudo apt-get install python-certbot-nginx -y
+  sudo certbot --nginx -d $WEBSITE_NAME --noninteractive --agree-tos --email $ADMIN_EMAIL --redirect
+  sudo service nginx reload
+  echo "SSL/HTTPS is enabled!"
+else
+  echo "SSL/HTTPS isn't enabled due to choice of the user!"
```
Comment (Yenthe666) [inline odoo_install.sh]: Since you also have a check for `[ $ADMIN_EMAIL != 'odoo@example.com' ]` this else might give invalid text.
Maybe split it up in another  condition or make the message say `SSL/HTTPS isn't enabled due to choice of the user or because of a misconfiguration!`

File: odoo_install.sh [hunk 6]
```diff
@@ -358,4 +376,7 @@ echo "Password superadmin (database): $OE_SUPERADMIN"
 echo "Start Odoo service: sudo service $OE_CONFIG start"
 echo "Stop Odoo service: sudo service $OE_CONFIG stop"
 echo "Restart Odoo service: sudo service $OE_CONFIG restart"
+if [ $INSTALL_NGINX = "True" ]; then
+  echo "Nginx configuration at: /etc/nginx/sites-available/odoo"
```
Comment (Yenthe666) [inline odoo_install.sh]: `echo "Nginx configuration file: /etc/nginx/sites-available/odoo"`

**Comment by Yenthe666** [comment]
Some neatpicks to make it more clear/consistent :) Overall it looks good.

File: odoo_install.sh (hunk 4 above)
Comment (tonybenoy) [reply odoo_install.sh:42]: @Yenthe666 Unless nginx is true this being true wont affect any changes. It's always better to have https than not have it right. Also I can add another check on the website name not = '_'

File: README.md [hunk 7]
```diff
@@ -21,6 +21,11 @@ There are a few things you can configure, this is the most used list:<br/>
 ```OE_SUPERADMIN``` is the master password for this Odoo installation.<br/>
 ```INSTALL_NGINX``` is set to ```False``` by default. Set this to ```True``` if you want to install Nginx.<br/>
 ```WEBSITE_NAME``` Set the website name here for nginx configuration<br/>
+```WEBSITE_NAME``` Set the website name here for nginx configuration<br/>
+```ENABLE_SSL``` Set this to ```True``` to install [certbot](https://github.com/certbot/certbot) and configure nginx with https using a free Let's Encrypted certificate<br/>
+```ADMIN_EMAIL``` Email is needed to register for Let's Enctryped Registeration. Replace the default placeholder with an email of your organisation.<br/>
```
Comment (Yenthe666) [inline README.md]:
```suggestion
```ADMIN_EMAIL``` Email is needed to register for Let's Encrypt registration. Replace the default placeholder with an email of your organisation.<br/>
```

File: README.md [hunk 8]
```diff
@@ -21,6 +21,11 @@ There are a few things you can configure, this is the most used list:<br/>
...
+```INSTALL_NGINX``` and ```ENABLE_SSL``` must be set to ```True``` and the placeholder in ```ADMIN_EMAIL``` must be replaced with a valid email address for certbot installation<br/>
+  _By enabling SSL though certbot you agree to the following [policies](https://www.eff.org/code/privacy/policy)_ <br/>
```
Comment (Yenthe666) [inline README.md]:
```suggestion
  _By enabling SSL through Let's Encrypt you agree to the following [policies](https://www.eff.org/code/privacy/policy)_ <br/>
```

File: odoo_install.sh (hunk 4 above)
Comment (Yenthe666) [reply odoo_install.sh:42]: okay sounds good :)

**Comment by Yenthe666** [comment]
Could you also split out the uninstall into another PR please?

**Comment by tonybenoy** [comment]
@Yenthe666 wasnt supposed to be a part of this PR. It isn't even complete. Note to self don't  use vs code git :p

**Comment by tonybenoy** [comment]
@Yenthe666 were you able to test this out?

**Comment by Yenthe666** [comment]
Thank you very much @tonybenoy! 🎉

**Comment by sel-carlitos** [comment]
I just tested the script on commit [cbe5387](https://github.com/Yenthe666/InstallScript/commit/cbe53879f379be35be16e3110bc177e83b99f507)

I get an error at line 164 which reads like so: 

`if [ $OE_VERSION >= "12.0" ];then
`
There is a space missing after the ; which I added to my downloaded script but still get the following error: 

```
---- Setting permissions on home folder ----
* Create server config file
* Creating server config file
./odoo_install.sh: line 164: [: 13.0: unary operator expected
* Create startup file
* Create init file
* Security Init File
* Start ODOO on Startup
```

**Comment by Yenthe666** [comment]
@izotecarlitos could you try again with this as code:
`if [[ "$OE_VERSION" >= "12.0" ]]; then`

**Comment by sel-carlitos** [comment]
No. It didn't work that way

image

I will try like this:

`if [[ $OE_VERSION >= "12.0" ]]; then`

**Comment by sel-carlitos** [comment]
No. it didn't like it that way either

image

**Comment by sel-carlitos** [comment]
It worked now! Like this: 

`if [[ "$OE_VERSION" > "11.0" ]]; then`

image


I think it needs to say 11.0 (instead of 12.0) as I understand that xmlrpc_port was up to v11? In other words from v12 onwards it is http_port?

**Comment by Yenthe666** [comment]
Hmm, it might because you've dropped the '=' operator actually.

**Comment by Yenthe666** [comment]
Should be fixed in https://github.com/Yenthe666/InstallScript/commit/62ad7239d012feb35c61d286a9679b18bbd9d707, thanks for reporting it!
